src/api/search/src/search.js

```diff
@@ -103,4 +101,57 @@ const search = async (options) => {
     values: hits.hits.map(({ _id }) => ({ id: _id, url: `${POSTS_URL}/${_id}` })),
   };
 };
-module.exports = { search };
+
+/**
+ * authorAutocomplete allows for querying author autocomplete fields
+ * @param { author } - text to search in author field
+ * @return all the results matching the fields text
+ */
+const authorAutocomplete = async ({ author }) => {
+  const results = {
+    query: {
+      match: {
+        'author.autocomplete': {
+          query: author,
+          operator: 'and',
+        },
+      },
+    },
+    highlight: {
+      fields: {
+        'author.autocomplete': {},
+      },
+    },
+  };
+
+  const {
+    body: { hits },
+  } = await client.search({
+    size: 10000,
+    _source: ['author'],
+    index,
+    body: results,
+  });
+
+  // Filter through all authors and remove duplicates
```

Comment:
Do we store duplicates?

If so, a faster way to do this is to put the data into a `Set` and then return an Array from that, since Sets don't include duplicates:

```js
const authors = [...new Set(hits.hits)];
```

Reply:
I'll try to answer the "size", "duplicate", and "highlights" conversation here.
Below is what a response from an ES Search request looks like. 
(I am using sample data on Kibana)
```js
{
  "took" : 2,
  "timed_out" : false,
  "_shards" : {
    "total" : 1,
    "successful" : 1,
    "skipped" : 0,
    "failed" : 0
  },
  "hits" : {
    "total" : {
      "value" : 3,
      "relation" : "eq"
    },
    "max_score" : 0.28194553,
    "hits" : [
      {
        "_index" : "posts",
        "_id" : "2",
        "_score" : 0.28194553,
        "_source" : {
          "author" : "Dave Humph"
        },
        "highlight" : {
          "author.autocomplete" : [
            "<em>Da</em>ve <em>H</em>umph"
          ]
        }
      },
      {
        "_index" : "posts",
        "_id" : "3",
        "_score" : 0.28194553,
        "_source" : {
          "author" : "Dave Humph"
        },
        "highlight" : {
          "author.autocomplete" : [
            "<em>Da</em>ve <em>H</em>umph"
          ]
        }
      },
      {
        "_index" : "posts",
        "_id" : "1",
        "_score" : 0.24156076,
        "_source" : {
          "author" : "David Humphrey"
        },
        "highlight" : {
          "author.autocomplete" : [
            "<em>Da</em>vid <em>H</em>umphrey"
          ]
        }
      }
    ]
  }
}
```

The "duplicates" are from the posts, cause each author could have more than one post. 
So if we set the `size` to 10, there's a possibility that the first 10 hits from ElasticSearch are all from the same one author.
Right now its set to the maximum amount of hits that it can page through so we make sure we get all `authors` that match the searched query, and then we take out the duplicates. Currently ElasticSearch doesn't have a way of knowing they are "duplicates", because for ES, they are different results from different "posts".

We won't really have this problem if we were autocompleting on another field with more unique text, for example the `title` of each post. 

Currently the `authors` and `highlights` array are from two different properties (`_score` and `highlight`) in the `hits.hits` array. But, since they're technically the same results, just one with `<em>...</em>` tags and one without, I'm assuming we only want to keep one of them then?

I don't think `Set` will work with `hits.hits` since all the `_id` for each post is different, and so will some `_score`. The `_score` is how ElasticSearch determines how relevant the result is to the query.

Reply:
Understood, thanks for clarifying and educating me.